### PR TITLE
Return error when no nameservers in resolv.conf

### PR DESCRIPTION
**Pull request by dav1do**
Addresses #2042. Raise an error instead of logging a warning if there isn't a nameserver in the resolv.conf and dns queries will never succeed.

### CODE REVIEW

File: crates/resolver/src/system_conf/unix.rs
Comment (djc) [inline crates/resolver/src/system_conf/unix.rs:90]: Nit: I think prevalent style is to all-lowercase error messages like this.

**Review by djc — APPROVED** [review]
This makes sense to me, thanks!

File: crates/resolver/src/system_conf/unix.rs
Comment (dav1do) [reply crates/resolver/src/system_conf/unix.rs:90]: Thanks for the quick review! So, I was going to leave it lowercase but I made it uppercase because the [error message below](https://github.com/hickory-dns/hickory-dns/pull/2327/files#diff-bd207ac182060e83598dd8cd393b88bfbcb4cca78c11b42583ddec40d46c0cc9R105) was capitalized. Let me know if I should just change this one or both to be more consistent with the other errors.

```rust
search.push(Name::from_str_relaxed(search_domain).map_err(|e| {
            io::Error::new(
                io::ErrorKind::Other,
                format!("Error parsing resolv.conf: {e}"),
            )
        })?);
```

Comment (dav1do) [reply crates/resolver/src/system_conf/unix.rs]: For now I just changed this one but am happy to update the other one too.

**Review by bluejekyll — APPROVED** [review]
looks good. Thank you!

**Comment by dav1do** [comment]
> looks good. Thank you!

Thank you, appreciate the quick turn around! Looks like I need workflow approval before I can merge so if one of you is able to kick those off again I'd be grateful.

**Pull request merged**
